chalice/cli/__init__.py

```diff
@@ -57,8 +58,29 @@ def create_new_project_skeleton(project_name, profile=None):
         f.write(GITIGNORE)
 
 
+def get_system_info(python_version_arg=None):
+    # type: (Optional[str]) -> str
+    if python_version_arg:
+        python_version = python_version_arg
+    else:
+        python_version = "{}.{}.{}".format(sys.version_info[0],
+                                           sys.version_info[1],
+                                           sys.version_info[2])
+    platform_system = platform.system()
+    platform_release = platform.release()
+    platform_info = "{}/{}".format(platform_system, platform_release)
+    system_info = "python {}, {}".format(python_version, platform_info)
+    if not(python_version.startswith("2.7.") or
+            python_version.startswith("3.6.")):
+        system_info += "\nWarning: Not using the recommended python version. "
```

Comment:
While I'm in favor of warning the user if they're using a version of python that isn't supported in lambda, I prefer we don't do this in the `--version` output.